src/linter.js

```diff
@@ -629,4 +633,23 @@ export default class Linter {
       });
   }
 
+  _markBadwordUsage(filename, fileData) {
+    if (fileData) {
+      const words = fileData.split(' ');
+
+      for (let word of words) {
+        // Filter out none a-Z characters to simplify matching with our
+        // badword list.
+        word = word.toLowerCase().replace(/[^a-zA-Z]/, '');
```

Comment:
This only normalizes words from the code-file so that we can easier match them with the badwords regular expression. All words in the badwords json file are `a-Z` only for now. I want to avoid doing needless matches for terms like `{` and stuff. I just pushed 24b0f48 that implements that.